### DISCUSSION TITLE
sbt migration from 0.13.0 to 1.3.0

### DISCUSSION THREAD

**Discussion by Nikitha963** [scalameta/metals#2740] [Q&A]
I am getting error while migrating sbt from 0.13.0 to 1.3.0. I am currently facing issue in error: not found: value scriptClasspath.

My build.sbt file after the migration.

val main = (project in file(".")).
     settings(
        appName         = "polaris",
        appVersion      = "1.strong text8.8",
        //scriptClasspath := Seq("modules/*", "customer-modules/*")
       // scriptClasspath = Seq[File] = file("modules/*") :: ("customer-modules/*") :: Nil
        **scriptClasspath** ~= { cp => cp.+:("modules/*").+:("customer-modules/*") }
    ).dependsOn(
       core, addressbook, pbx, pbxAppSoftphones, pbxAppCallLog, pbxAppQueues, pbxAppPhonebook, pbxAppClick2dial, pbxAppOperator
    ).aggregate(
       core, addressbook, pbx, pbxAppSoftphones, pbxAppCallLog, pbxAppQueues, pbxAppPhonebook, pbxAppClick2dial, pbxAppOperator
    )
and also I have attached plugin.sbt file below -

// Comment to get more information during initialization
logLevel := Level.Warn

// The Typesafe repository
//resolvers += "Typesafe repository" at "https://repo.typesafe.com/typesafe/releases/"

//resolvers += "Maven Central Server" at "https://repo1.maven.org/maven2"

resolvers += Resolver.url("bintray-sbt-plugins", url("http://dl.bintray.com/sbt/sbt-plugin-releases"))(Resolver.ivyStylePatterns)

// Use the Play sbt plugin for Play projects
addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.7.2")

addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.10")

addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.7.6")
What wrong am I doing while migration? there is a syntax error I am not able to figure out.

 plugins.sbt and build.sbt files are under project folder. scriptClasspath comes from build.sbt

I followed the below migration guide and changed file name from Build.scala to build.sbt 

 scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html

**Comment by tgodzik** [comment]
It doesn't look like there is anything like `scriptClasspath` anymore, just `scriptedClasspath`. Have you tried asking on https://gitter.im/sbt/sbt ?

**Reply by Nikitha963** [reply]
Hi @tgodzik I will check with scriptedClasspath and try to check whether I get the same error.

Ya I created a room for this as below : 

https://gitter.im/Nikitha963/community#

**Reply by tgodzik** [reply]
Best to ask in the sbt channel itself, more people might be available to help you. Eugene might be too busy to help out especially that he only does it in his free time.

**Reply by Nikitha963** [reply]
Hi @tgodzik I am very new to scala and also sbt. I have mentioned my query in the one which you mentioned.

https://gitter.im/sbt/sbt and also I tried with scriptedClasspath now I am getting different error :

error: value +: is not a member of sbt.io.PathFinder
	scriptedClasspath ~= { cp => cp.+:("modules/*").+:("customer-modules/*") }

Checking on this currently. Thank you so much

**Reply by Nikitha963** [reply]
@tgodzik I added in https://gitter.im/sbt/sbtbut not getting any reply. what should we give instead of  "+:" in below line :

 scriptedClasspath ~= { cp => cp.+:("modules/").+:("customer-modules/") }

**Reply by laughedelic** [reply]
Try changing it either to 

```scala
scriptedClasspath ~= { cp => cp :+ "modules/" :+ "customer-modules/" }
```

or to 

```scala
scriptedClasspath ~= { cp => "modules/" +: "customer-modules/" +: cp }
```

`:+` adds elements on the right, `+:` adds elements on the left.

**Reply by Nikitha963** [reply]
@laughedelic I tried the above two ways still the same issue

error: value :+ is not a member of sbt.io.PathFinder
	scriptedClasspath ~= { cp => cp :+ "modules/" :+ "customer-modules/" }
                                        ^
error: value +: is not a member of sbt.io.PathFinder
	scriptedClasspath ~= { cp => "modules/" +: "customer-modules/" +: cp }
                                                                       ^
 error: type mismatch;
 found   : scala.collection.immutable.IndexedSeq[Any]
 required: sbt.io.PathFinder
	scriptedClasspath ~= { cp => cp +: "modules/" +: "customer-modules/" }
                                        ^
can you please help me to fix this

**Reply by Nikitha963** [reply]
@laughedelic and @tgodzik how to proceed can you help me in resolving this scriptClasspath issue.

**Reply by Nikitha963** [reply]
@laughedelic and @tgodzik its working fine now. my project structure is wrong so I was facing this issue. Thanks

**Reply by Nikitha963** [reply]
Hi , after changing the project structure and run sbt console : still facing some issues and also scriptClasspath issue :(

error] References to undefined settings:
[error]
[error] core / buildInfo from core / Compile / sourceGenerators (/home/nikitha/Documents/Binastar/Polaris/polaris_sbt_upgrade_1.3.0/polaris-upgrade-sbt/modules/core/build.sbt:32)
[error]
[error] scriptClasspath from scriptClasspath (/home/nikitha/Documents/Binastar/Polaris/polaris_sbt_upgrade_1.3.0/polaris-upgrade-sbt/build.sbt:50)
[error]

**Reply by laughedelic** [reply]
@Nikitha963 this really doesn't belong here. let's move this to your [SO question](https://stackoverflow.com/q/67232850/736957).
